test-project/settings.gradle.kts

```diff
@@ -0,0 +1,8 @@
+pluginManagement {
+    includeBuild("..")
+}
+dependencyResolutionManagement {
+    repositories {
+        mavenCentral()
+    }
+}
```

Comment:
That's the new feature in Gradle 7.0 👌 https://docs.gradle.org/7.0/release-notes.html#plugin-development-improvements
I left a few commits that contain pre-7.0 approach (which publish the plugin to `mavenLocal` and than applies it to a test projec). I decided to _leave them_, just in case, if for some reason we can't use the new feature.